### ISSUE TITLE
UX of Refresh Waterfox

### ISSUE THREAD

**Issue by grahamperrin** [BrowserWorks/waterfox#562]
From [Refresh Waterfox: graceful handling of extensions.getAddons.cache.enabled and its cached data · Issue #392 …](https://github.com/MrAlex94/Waterfox/issues/392): 

> … a reminder for me to raise a pull request for a minor inconsistency with the wording of the Waterfox refresh routine.

– that, and some other things, one or two of which have [paper cut](https://en.wikipedia.org/wiki/Paper_cut_bug) qualities. 

I'll not have any part of the PR in time for release of 56.2.0 (sorry), maybe later this month. 

Meta, tracking: #538

**Comment by grahamperrin** [comment]
56.2.0, build 20180514223240: 

![2018-05-15 22 10 53 refresh waterfox](https://user-images.githubusercontent.com/192271/40084624-a5ed37ae-588f-11e8-8470-7cd5f7fc83bb.png)

Issues include: 

- third line of text missing, or invisible (wonder whether my PR a few weeks ago inadvertently wrote too much into the available space)
- without that third line, which ended with a colon `:` its not clear that the indented lines should have been bullet points (someone who's unfamiliar with the routine might imagine that the button will lead to a wizard for each of the five things, maybe with options to skip)
- re: #370 metadata is _not_ removed (a good thing, one less warning to include in the dialogue). 

Also, for reference: 

- [Refresh Waterfox: appearance add-ons (themes) are removed · Issue #391 …](https://github.com/MrAlex94/Waterfox/issues/391)

**Comment by grahamperrin** [comment]
I vaguely recall making a mess (with a PR) of one or two of the lines that appear near the top right of about:support … at a glance it's no longer a mess (thanks to whoever made good), now I see there's another **Refresh Waterfox** dialogue with slightly different wording: 

![2018-05-15 23 04 40 refresh waterfox](https://user-images.githubusercontent.com/192271/40086017-7ff200a2-5894-11e8-85de-a119cb758ac6.png)

The two identically titled dialogues should be made consistent with each other.

**Comment by grahamperrin** [comment]
From one of the two dialogues: 

> … restore performance. …

I wasn't explicit at <https://github.com/MrAlex94/Waterfox/pull/457/files#diff-3dddd4c23c5710a43d34a46de3f07e54R6> … an alternative phrase was chosen because a refresh of Waterfox _does not always_ lead to improved performance. 

tl;dr (maybe contentious … I can expand upon this at a later date, when things are less busy): 

- emphases on **minimising problems, and risks of problems** – and on **healing** (e.g. where a user might have shot himself, or herself, in the foot at about:config)

– emphases more on things such as those, than on sating appetites for speed.

----

From <https://github.com/MrAlex94/Waterfox/pull/457#issue-172661930> (merged): 

> - documentation.

If there's to be an FAQ page – <https://waterfoxproject.org/faq>, maybe – then a section of the page can answer questions about the refresh routine. 

(Copy for the page can be drafted elsewhere, not in this issue.)

**Comment by grahamperrin** [comment]
From <https://github.com/MrAlex94/Waterfox/issues/380#issuecomment-397793991>: 

> - multiprocess windows reportedly disabled forcibly

**Comment by grahamperrin** [comment]
Shots from 56.2.1 on Mac OS X 10.9.5: 

![2018-06-16 07 36 41](https://user-images.githubusercontent.com/192271/41496908-4e496722-7142-11e8-9e0f-42949c430821.png)

![2018-06-16 07 37 51](https://user-images.githubusercontent.com/192271/41496909-52070e1e-7142-11e8-93de-59597e606ce4.png)

![2018-06-16 07 38 05](https://user-images.githubusercontent.com/192271/41496910-576f5d70-7142-11e8-93f0-09bda9642feb.png)

![2018-06-16 07 38 13](https://user-images.githubusercontent.com/192271/41496914-71ddddb2-7142-11e8-8c0a-85b3f219b4a3.png)

![2018-06-16 08 21 28](https://user-images.githubusercontent.com/192271/41496915-764bc5f8-7142-11e8-8207-4c3231e772de.png)

**Comment by grahamperrin** [comment]
tl;dr from 632: 

![welcome-back](https://user-images.githubusercontent.com/192271/41496804-432f41a6-7140-11e8-95ab-5f79a28f1ff3.png)
